diff --git a/vendor/Squidiff/Squidiff/scrna_datasets.py b/vendor/Squidiff/Squidiff/scrna_datasets.py
index b6386c2..feeac82 100644
--- a/vendor/Squidiff/Squidiff/scrna_datasets.py
+++ b/vendor/Squidiff/Squidiff/scrna_datasets.py
@@ -48,15 +48,15 @@ class AnnDataDataset(Dataset):
                 self.control_features = torch.tensor(control_adata.X, dtype=torch.float32)
             else:
                 self.control_features = torch.tensor(control_adata.X.toarray(), dtype=torch.float32)
-                
+
             self.drug_type_list = adata.obs['SMILES'].to_list()
             self.dose_list = adata.obs['dose'].to_list()
             #self.encoded_obs_tensor = torch.tensor(adata.obs['Group'].copy().values, dtype=torch.float32)
-            self.encoded_obs_tensor = adata.obs['Group'].copy().values
+            self.encoded_obs_tensor = torch.tensor(adata.obs['Group'].copy().values, dtype=torch.float32)
             self.encode_drug_doses = Drug_dose_encoder(self.drug_type_list, self.dose_list, comb_num=comb_num)
             self.encode_drug_doses = torch.tensor(self.encode_drug_doses, dtype=torch.float32)
         else:
-            self.encoded_obs_tensor = adata.obs['Group'].copy().values
+            self.encoded_obs_tensor = torch.tensor(adata.obs['Group'].copy().values, dtype=torch.float32)
         
     def __len__(self):
         return len(self.features)
